konduit-serving-core/src/main/java/ai/konduit/serving/pipeline/steps/BertStepRunner.java

```diff
@@ -0,0 +1,152 @@
+ *  *  * terms of the Apache License, Version 2.0 which is available at
+ *  *  * https://www.apache.org/licenses/LICENSE-2.0.
+ *  *  *
+ *  *  * Unless required by applicable law or agreed to in writing, software
+ *  *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  *  * License for the specific language governing permissions and limitations
+ *  *  * under the License.
+ *  *  *
+ *  *  * SPDX-License-Identifier: Apache-2.0
+ *  *  *****************************************************************************
+ *
+ *
+ */
+
+package ai.konduit.serving.pipeline.steps;
+
+import ai.konduit.serving.config.SchemaType;
+import ai.konduit.serving.pipeline.PipelineStep;
+import ai.konduit.serving.pipeline.step.BertStep;
+import ai.konduit.serving.util.WritableValueRetriever;
+import io.vertx.core.json.JsonObject;
+import org.datavec.api.records.Record;
+import org.datavec.api.writable.Writable;
+import org.deeplearning4j.iterator.BertIterator;
+import org.deeplearning4j.iterator.LabeledSentenceProvider;
+import org.deeplearning4j.iterator.provider.CollectionLabeledSentenceProvider;
+import org.deeplearning4j.nn.graph.ComputationGraph;
+import org.deeplearning4j.text.tokenization.tokenizerfactory.BertWordPieceTokenizerFactory;
+import org.deeplearning4j.util.ModelSerializer;
+import org.nd4j.base.Preconditions;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.datavec.api.writable.Text;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class BertStepRunner extends BaseStepRunner
+{
+    private BertWordPieceTokenizerFactory tokenizer;
+    private BertStep bertStep;
+
+    private ComputationGraph bertModel;
+
+    private final int MAX_LEN = 256;
+
+    public BertStepRunner(PipelineStep pipelineStep)
+    {
+        super(pipelineStep);
+
+        this.bertStep = (BertStep) pipelineStep;
+
+        //String inputName = (String) this.bertStep.getInputNames().get(0);
+        //String outputName = (String) this.bertStep.getOutputNames().get(0);
+
+        String modelPath = (String) this.bertStep.getModelPath();
+        String vocabPath = (String) this.bertStep.getVocabPath();
+
+        Map<String, SchemaType[]> inputSchema = this.bertStep.getInputSchemas();
+
+        //load tokenizer
+        try {
+
+            this.tokenizer = new BertWordPieceTokenizerFactory(new File(vocabPath), true, true, StandardCharsets.UTF_8);
+
+        } catch (IOException e)
+        {
+            throw new IllegalStateException("Vocabulary file missing");
+        }
+
+        //load Bert trained model
+        try {
+
+            this.bertModel = ModelSerializer.restoreComputationGraph(modelPath);
+
+        } catch (IOException e)
+        {
+            throw new IllegalStateException("Model file failed loading");
+        }
```

Comment:
Wrap the existing exception in the new exception, it could contain important information for debugging:
```
        try {
            this.bertModel = ModelSerializer.restoreComputationGraph(modelPath);
        } catch (IOException e) {
            throw new IllegalStateException("Failed to load ComputationGraph model at path " + modelPath.getAbsolutePath(), e);
        }
```